core/conversion/converters/impl/interpolate.cpp

```diff
@@ -0,0 +1,113 @@
+#include "torch/torch.h"
+#include "core/util/prelude.h"
+#include "core/conversion/converters/converters.h"
+
+#include <csignal>
+
+namespace trtorch {
+namespace core {
+namespace conversion {
+namespace converters {
+namespace impl {
+namespace {
+
+auto interpolate_registrations = RegisterNodeConversionPatterns()
+    .pattern({
+        "aten::upsample_nearest1d(Tensor self, int[1] output_size, float? scales=None) -> (Tensor)",
+        [](ConversionCtx* ctx, const torch::jit::Node*n, args& args) -> bool {
+            auto in = args[0].ITensor();
+            auto in_shape = util::toVec(in->getDimensions());
+
+            // Case 1: user uses output size and not scales
+            if (!args[1].IValue()->isNone() && args[2].IValue()->isNone()) {
+                auto out_size = util::toVec(util::toDims(args[1].unwrapToIntList()));
+
+                TRTORCH_ASSERT(out_size.size() == 1, "aten::upsample_nearest1d input Tensor and output size dimension mismatch");
+                
+                auto out_shape = in_shape;
+                std::copy(out_size.begin(), out_size.end(), out_shape.begin() + (in_shape.size() - out_size.size()));
+
+                auto resize_layer = ctx->net->addResize(*in);
+                TRTORCH_CHECK(resize_layer, "Unable to create interpolation (resizing) layer from node" << *n);
+
+                resize_layer->setOutputDimensions(util::toDims(out_shape));
+                resize_layer->setResizeMode(nvinfer1::ResizeMode::kNEAREST);
+                resize_layer->setName(util::node_info(n).c_str());
+
+                auto layer_output = ctx->AssociateValueAndTensor(n->outputs()[0], resize_layer->getOutput(0));
+                LOG_DEBUG("Output tensor shape: " << layer_output->getDimensions());
+            } else {
+                LOG_DEBUG("scale factor parameter not supported yet.");

```

Comment:
I feel like this should either be a warning or a full error out, probably an error since the program shouldn't compile without this layer. Here is an example on how to throw an error

```
 TRTORCH_THROW_ERROR("Unable to retrieve all node inputs for node: " \
                                << util::node_info(n) << " (ctx.AddLayer)\nSpecifically failed to retrieve value for input: " \
                                << *input_node);
```